app/browser/reducers/tabsReducer.js

```diff
@@ -367,8 +436,10 @@ const tabsReducer = (state, action, immutableAction) => {
         }
         break
       }
-    case appConstants.APP_FRAME_CHANGED:
-      state = tabState.updateFrame(state, action, shouldDebugTabEvents)
+    case appConstants.APP_FRAMES_CHANGED:
+      for (const frameAction of action.get('frames').valueSeq()) {
```

Comment:
might be good to check if `frames` is truthy (ex: `action.has('frames')`) before using

Reply:
This action does not get called without that property. Should we have a policy of trusting our actions in the way that they are currently being called, or optimize for possible future mistakes?